pkg/workflows/management/install_new_components.go

```diff
@@ -9,21 +9,28 @@ import (
 
 type installNewComponents struct{}
 
-// Run installNewComponents performs actions needed to upgrade the management cluster.
-func (s *installNewComponents) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+func runInstallNewComponents(ctx context.Context, commandContext *task.CommandContext) error {
```

Comment:
nit: same here
```suggestion
func installNewComponents(ctx context.Context, commandContext *task.CommandContext) error {
```

Reply:
same, the installNewComponents name has been take by the struct

Reply:
for what are you optimizing by having this function at all? Is it redundant to `func (s *installNewComponents) Run`?